src/utils/FSHLogger.ts

```diff
@@ -147,8 +147,8 @@ class LoggerStats {
 export const stats = new LoggerStats();
 
 export class ErrorsAndWarnings {
-  public errors: { message: string; location?: string; input?: string }[] = [];
-  public warnings: { message: string; location?: string; input?: string }[] = [];
+  public errors: { message: string; location?: any; input?: string }[] = [];
+  public warnings: { message: string; location?: any; input?: string }[] = [];
```

Comment:
I looked into this type for `location` a bit because it seemed odd to me that the type would be so off based on the examples you pointed out in issue #1106. It looks like in practice, `location` will always be of the type `TextLocation`, which is defined [here](https://github.com/FHIR/sushi/blob/master/src/fshtypes/FshEntity.ts#L53-L58). This actually matches the type that is checked in the second test you called out in the issue. Instead of changing this type to `any`, I would suggest changing it to `TextLocation` here. Then I would also suggest updating the first test you called out in the issue to update the `location` property that is used and checked there. Instead of using

```
location: [1, 2, 3, 4]
```

you can transform that array to a text location like 
```
location: {
  startLine: 1,
  startColumm: 2,
  endLine: 3,
  endColumn: 4
}
```
If you replace the `location` in the test ([here](https://github.com/FHIR/sushi/blob/master/test/utils/FSHLogger.test.ts#L86-L103)) where it is used in the `logger.warn` call and in the expect statement, I think the test will be represent what happens in the code.